nntrainer/optimizers/adam.cpp

```diff
@@ -26,17 +26,8 @@ const std::string Adam::type = "adam";
 
 enum AdamParams { wm, wv };
 
-void Adam::addOptimizerVariable(std::vector<Weight> &weight_list) {
-  for (auto &w : weight_list) {
-    w.clearOptimizerVariables();
-
-    // TODO: only trainable weights must be sent to optimizer
-    if (!w.getTrainable())
-      continue;
-
-    w.addOptimizerVariable(w.getDim()); /** Add wm */
-    w.addOptimizerVariable(w.getDim()); /** Add wv */
-  }
+std::vector<TensorDim> Adam::addOptimizerVariable(const TensorDim &dim) {
+  return {dim, dim};
```

Comment:
I guess the function name is not valid anymore. It returns just vector of dimensions.. So.. Could you consider change the name?

Reply:
rename addOptimizerVariable to getOptimizerVariableDim in optimizer :)